lib/utils/signature.js

```diff
@@ -0,0 +1,395 @@
+/*!
+ * signature.js - helper class for signature.
+ * Copyright (c) 2019, Nodari Chkuaselidze (MIT License)
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const secp256k1 = require('bcrypto/lib/secp256k1');
+const common = require('../protocol/common');
+
+/**
+ * Size of the curve
+ * @const {Number}
+ */
+
+const SIZE = 32;
+
+/**
+ * 32 byte number zero
+ * @const {Buffer}
+ */
+
+const ZERO = Buffer.alloc(32, 0x00);
+
+/**
+ * This based on bcrypto/lib/internal/signature
+ * Adds recid on top.
+ * @property {Buffer} r
+ * @property {Buffer} s
+ * @property {Number} recid
+ */
+
+class LedgerSignature {
+  /**
+   * Create LedgerSignature object.
+   *
+   * @param {Buffer?} r
+   * @param {Buffer?} s
+   * @param {Number?} recid
+   */
+
+  constructor(r, s, recid) {
+    this.r = ZERO;
+    this.s = ZERO;
+    this.recid = -1;
+
+    if (r)
+      this.setR(r);
+
+    if (s)
+      this.setS(s);
+
+    if (recid)
+      this.setRecid(recid);
+  }
+
+  /**
+   * Inject options
+   * @param {Object} options
+   * @param {Buffer?} options.r
+   * @param {Buffer?} options.s
+   * @param {Number?} options.recid
+   * @returns {LedgerSignature}
+   */
+
+  fromOptions(options) {
+    assert(typeof options === 'object');
+
+    if (options.r)
+      this.setR(options.r);
+
+    if (options.s)
+      this.setS(options.s);
+
+    if (options.recid)
+      this.setRecid(options.recid);
+
+    return this;
+  }
+
+  /**
+   * Set R.
+   * @param {Buffer} r
+   * @returns {LedgerSignature}
+   */
+
+  setR(r) {
+    assert(Buffer.isBuffer(r));
+    assert(r.length === SIZE);
+
+    this.r = r;
+
+    return this;
+  }
+
+  /**
+   * Set S.
+   * @param {Buffer} s
+   * @returns {LedgerSignature}
+   */
+
+  setS(s) {
```

Comment:
NIT: It seems like it would be more natural to use the builtin javascript getter/setter. Not super opinionated on this since it works already
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Defining_getters_and_setters